### ISSUE TITLE
UI server is blocking graceful application shutdown

### ISSUE THREAD

**Issue by afpr252** [clojure-goes-fast/clj-async-profiler#25]
While developing my application that uses this library I noticed that the ui-server is blocking the shutdown of the application. It would be nice to have a function to stop the UI HTTP server in the core namespace so that the application can gracefully shutdown.

I'm happy to open a PR with the necessary changes.

**Comment by alexander-yakushev** [comment]
Hello, André! I could indeed reproduce this behavior. The problem comes from the server handler thread being non-daemon (hence, it prevents the application from shutting down). It makes more sense for handler threads to be daemon threads.

Would you still care for an explicit stop-function if your direct issue was resolved through the daemon threads?

**Comment by afpr252** [comment]
If it becomes a daemon thread I no longer need an explicit stop function.

Thank you for looking into this so quickly.

**Comment by alexander-yakushev** [comment]
Change of plans. Apparently, providing an executor for HttpServer does not change the type of thread that is created for dispatching, and it is explicitly non-daemon.

I have added a function `clj-async-profiler.ui/stop-server` and pushed it as `1.0.3-SNAPSHOT`. Would this be enough? Can you check if this unblocks the shutdown?

**Comment by afpr252** [comment]
I think this would be enough, yes. I will test it later today. Thanks

**Comment by afpr252** [comment]
I just tested and using `clj-async-profiler.ui/stop-server` works.

**Comment by alexander-yakushev** [comment]
Deployed as 1.0.3. Thank you!
